graphstorm-processing/graphstorm_processing/data_transformations/dist_feature_transformer.py

```diff
@@ -56,7 +62,9 @@ def __init__(self, feature_config: FeatureConfig):
         elif feat_type == "bucket-numerical":
             self.transformation = DistBucketNumericalTransformation(**default_kwargs, **args_dict)
         elif feat_type == "categorical":
-            self.transformation = DistCategoryTransformation(**default_kwargs, **args_dict)
+            self.transformation = DistCategoryTransformation(
+                **default_kwargs, **args_dict, spark=spark
```

Comment:
Why we need spark session here, but for others we dont.

Reply:
To save the transformation representation, we need to be able to create Spark DataFrames and we need a SparkContext to do so. So far the other transformations do not need one.